lib/rspec/core/formatters/base_formatter.rb

```diff
@@ -54,16 +55,23 @@ def close(_notification)
       private
 
         def start_sync_output
-          @old_sync, output.sync = output.sync, true if output_supports_sync
+          if output_supports_sync
+            @sync_started = true
+            @old_sync, output.sync = output.sync, true
+          end
         end
 
         def restore_sync_output
-          output.sync = @old_sync if output_supports_sync && !output.closed?
+          output.sync = @old_sync if sync_started? && !output.closed?
```

Comment:
no need for the check on `output_supports_sync` because `@sync_started` is only set to true within the `output_supports_sync` check in `start_sync_output`

Reply:
This should set `@sync_started` to false afterwards